### PR TITLE
chore: ruff ANN401

### PR DESCRIPTION
**Pull request by rokam**
## Summary by CodeRabbit

- **New Features**
  - Enhanced type safety across various device modules by updating the `set_attribute` method to accept specific types (`int`, `str`, `bool`).
  - Added a new `ValueWrongType` exception for handling incorrect data type errors.

- **Bug Fixes**
  - Improved attribute value handling in Midea A1, AC, B3, B4, x26, x34, and x40 device modules to ensure correct type conversions and settings.

- **Tests**
  - Updated `test_set_attribute` in `device_ac_test.py` to reflect changes in method calls and type validation.

- **Refactor**
  - Added type hints and annotations to various methods and attributes for better code clarity and maintainability.

### CODE REVIEW

File: midealocal/device.py [hunk 1]
```diff
@@ -554,7 +554,7 @@ def run(self) -> None:
                     self.close_socket()
                     break
 
-    def set_attribute(self, attr: str, value: Any) -> None:
+    def set_attribute(self, attr: str, value: Any) -> None:  # noqa: ANN401
```
Comment (chemelli74) [inline midealocal/device.py:557]:
```suggestion
    def set_attribute(self, attr: str, value: bool | int | str) -> None: 
```

File: midealocal/devices/b0/__init__.py [hunk 2]
```diff
@@ -100,7 +100,7 @@ def process_message(self, msg: bytes) -> dict:
                 new_status[str(status)] = self._attributes[status]
         return new_status
 
-    def set_attribute(self, attr: str, value: Any) -> None:
+    def set_attribute(self, attr: str, value: Any) -> None:  # noqa: ANN401
```
Comment (chemelli74) [inline midealocal/devices/b0/__init__.py:103]:
```suggestion
    def set_attribute(self, attr: str, value: bool | int | str) -> None:
```

**Review by chemelli74 — CHANGES_REQUESTED** [review]
I see `set_attribute` has different signatures. Can we fix them ?

**Comment by rokam** [comment]
> I see `set_attribute` has different signatures. Can we fix them ?

I don't know if this is better then leaving the right scope in each device. That's why I left some with Any.

**Review by chemelli74 — APPROVED** [review]
Very nice one, thx !

**Pull request merged**